File: NumberGuessingGame/RandomNumberGuess.c
#include <stdio.h>
#include <stdlib.h> // Standard library for memory allocation, process control, conversions, and others

#include <time.h>

int calculateScore(int guessCounter) { // Function to calculate score based on the number of attempts

    switch (guessCounter) { // Determine score based on the number of guesses

        case 1: return 100;
        case 2: return 90;
        case 3: return 80;
        case 4: return 60;
        case 5: return 40;
        case 6: return 20;
        case 7: return 5;
        default: return 0;
    }
}

int main() {
    int numberToGuess, chances = 7, guessCounter = 0, myGuess, myScore;

    printf("Hi welcome to the game, This is a number guessing game.\n");
    printf("You got 7 chances to guess the number. Let's start the game\n");

    srand(time(NULL)); // Seed the random number generator with the current time

    numberToGuess = rand() % 100; // Generate a random number between 0 and 99 for the user to guess

    printf("%d\n", numberToGuess); //for debuging

    while (guessCounter < chances) { // Loop until the user has used all chances

        guessCounter++;
        printf("Please Enter your Guess(1 to 100): "); // Prompt user for their guess

        scanf("%d", &myGuess);

        if (myGuess == numberToGuess) {
            myScore = calculateScore(guessCounter);
            printf("The number is %d and you found it right !! in the %d attempt\n", numberToGuess, guessCounter); // Display the correct number and attempt

            printf("Your score is: %d\n", myScore); // Display the user's score

            break;
        } else if (guessCounter >= chances) {
            printf("Oops sorry, The number is %d better luck next time\n", numberToGuess);
            break;
        } else if ((myGuess > numberToGuess) && (myGuess < 101)) {
            printf("Your guess is higher \n"); // Feedback if the guess is too high

        } else if ((myGuess < numberToGuess) && (myGuess > 0)) {
            printf("Your guess is lesser\n"); // Feedback if the guess is too low

        } else{
          printf("Number out of bounds\n"); // Feedback if the guess is out of bounds

        }
    }

    return 0;
}

File: NumberGuessingGame/RandomNumberGuess.js
const readline = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});

// Welcome message
console.log("Hi welcome to the game, This is a number guessing game.");
console.log(
  "You have 7 chances to guess the number, which is between 0 and 99. Let's start the game!"
);

// Generating a random number between 0 and 99
const numberToGuess = Math.floor(Math.random() * 100);
const totalChances = 7; // Number of chances
let guessCounter = 0; // Counter for the number of guesses
let myScore = 0; // Score for each guess

function askForGuess() {
  if (guessCounter < totalChances) {
    guessCounter++;
    readline.question(
      `Please Enter your Guess (Attempt ${guessCounter}): `,
      (answer) => {
        const myGuess = parseInt(answer, 10);

        if (isNaN(myGuess) || myGuess < 0 || myGuess > 99) {
          console.log("Invalid input. Please enter a number between 0 and 99.");
          guessCounter--; // Decrement counter as it was an invalid guess
          askForGuess(); // Ask again
          return;
        }

        if (myGuess === numberToGuess) {
          switch (guessCounter) {
            case 1:
              myScore = 100;
              break;
            case 2:
              myScore = 90;
              break;
            case 3:
              myScore = 80;
              break;
            case 4:
              myScore = 60;
              break;
            case 5:
              myScore = 40;
              break;
            case 6:
              myScore = 20;
              break;
            case 7:
              myScore = 5;
              break;
            default:
              myScore = 0;
              break;
          }
          console.log(
            `Congratulations! The number is ${numberToGuess} and you guessed it correctly in attempt ${guessCounter}!`
          );
          console.log(`Your score is ${myScore} points.`);
          readline.close();
        } else if (myGuess > numberToGuess) {
          console.log("Your guess is higher.");
          console.log(
            `You have ${totalChances - guessCounter} chances remaining.`
          );
          askForGuess();
        } else if (myGuess < numberToGuess) {
          console.log("Your guess is lower.");
          console.log(
            `You have ${totalChances - guessCounter} chances remaining.`
          );
          askForGuess();
        }
      }
    );
  } else {
    console.log(
      `Oops, you've run out of chances! The number was ${numberToGuess}.`
    );
    console.log(`Your final score is ${myScore} points.`);
    readline.close();
  }
}

askForGuess();
